apps/transformer/lib/transform/add_timestamp.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Transform.AddTimestampField do
+  @moduledoc """
+  `Transform.Step` impl to add a timestamp field to a dataset. Timestamps will be stored in ISO-8601 format.
+
+  ## Configuration
+  * `name` - Required. Field name or path to store the timestamp under. Ex: `"timestamp"` or `["timestamp"]`
+  """
+  use Definition, schema: Transform.AddTimestampField.V1
+
+  @type t :: %__MODULE__{
+          name: String.t() | [String.t()]
+        }
+
+  defstruct name: nil,
+            description: ""
+
+  defimpl Transform.Step, for: __MODULE__ do
+    import Dictionary.Access, only: [to_access_path: 1]
+
+    def transform_dictionary(%{name: name, description: description}, dictionary) do
+      name_path = to_access_path(name)
+
+      put_in(
+        dictionary,
+        name_path,
+        Dictionary.Type.Timestamp.new!(%{
+          name: name,
+          description: description
+        })
+      )
+      |> Ok.ok()
+    end
+
+    def create_function(%{name: name}, _dictionary) do
+      Ok.ok(fn record ->
+        Map.put(record, name, DateTime.utc_now() |> DateTime.to_iso8601()) |> Ok.ok()
```

Comment:
are both of these `Ok.ok/1` necessary? It feels redundant to me unless I'm missing something.

Reply:
As I understand it, the create_function method returns a function as part of a tuple with the OK atom. That function, when acting on a record, returns another tuple with the OK atom and the new record made by calling the function. This is the same pattern followed by the delete and move field transforms.

Reply:
`create_function` returns a function in an `{:ok, fun}` tuple. Running that function returns its own `{:ok, return_value}` tuple.

I agree that it's a bit confusing, but that's the `Transformer.transform_function` type.

Reply:
Would it be easier to parse if we re-arranged?

```elixir
fn record -> 
  now = DateTime.utc_now() |> DateTime.to_iso8601()

  Map.put(record, name, now)
  |> Ok.ok()
end
|> Ok.ok()
```

Reply:
I believe this should be a NaiveDateTime.  During normalization all timestamps are converted to UTC and turned into a NaiveDateTime.to_iso8601.

Reply:
Did the rearrangement and switched to NaiveDateTime.

Reply:
> Would it be easier to parse if we re-arranged?
> 
> ```elixir
> fn record -> 
>   now = DateTime.utc_now() |> DateTime.to_iso8601()
> 
>   Map.put(record, name, now)
>   |> Ok.ok()
> end
> |> Ok.ok()
> ```

Yes, I think that helps slightly but the primary confusion is the "nested" `:ok` tuples; thanks for clarifying.